### DISCUSSION TITLE
当使用json-schema时 key如果为a[b] form表单的结构会变为a.b

### DISCUSSION THREAD

**Discussion by HilaryPeng** [alibaba/formily#2859] [Q&A 2.x] [answered]
![Screen Shot 2022-02-24 at 9 39 24 PM](https://user-images.githubusercontent.com/22334441/155535283-2103575d-02fe-44af-b297-d71d9da0c1a7.png)

**Comment by HilaryPeng** [comment]
https://codesandbox.io/s/quizzical-architecture-qs0rj8 
附上地址

**Comment by janryWang** [comment, marked as answer]
加转义吧，`\\`

**Reply by HilaryPeng** [reply]
由于数据结构为后台API返回的 所以key可能会出现这种情况。 现在每层的key都做了一个转译的处理。
但其实在以上场景中 "input[a]"这种结构更倾向于当一个字符串 特别是函数生成json结构的时候 
这点解析不知道是否可以优化下 下次遇到的小伙伴可以参考下
<img width="731" alt="Screen Shot 2022-02-25 at 6 26 08 PM" src="https://user-images.githubusercontent.com/22334441/155699247-3d7a8379-99b4-475a-b278-61cc46040fab.png">

**Reply by janryWang** [reply]
不会优化，中括号用法是属于feature
